server/services/query-memory-db/queries/business-process-rejection-reasons.js

```diff
@@ -0,0 +1,113 @@
+				id: null,
+				businessName: null
+			},
+			operator: {
+				id: null,
+				name: null
+			},
+			processingStep: {
+				path: null,
+				label: null
+			},
+			occurancesCount: 0,
+			date: null
+		};
+		businessProcess = db.BusinessProcess.getById(businessProcessId);
+		if (!businessProcess) return;
+		result.service.businessName = businessProcess.businessName;
+		result.service.id = businessProcessId;
+		result.service.type = businessProcess.constructor.__id__;
+		//rejectionReasons -> processingSteps
+		businessProcess.processingSteps.applicable.some(function (processingStep) {
+			otherValue = '';
+			if (!processingStep.isRejected && !processingStep.isSentBack) return;
+			processStep = processingStep;
+
+			result.rejectionType = processingStep.isRejected ? 'rejected' : 'sentBack';
+			if (processingStep.steps) { // processingStepGroup. We need the full path.
+				processingStep.steps.applicable.some(function (step) {
+					if (!step.isRejected && !step.isSentBack) return;
+					processStep = step;
+					return true;
+				});
+			}
+			result.processingStep.label = processStep.label;
+			result.processingStep.path = processStep.__id__.slice(processStep.__id__.indexOf('/'));
+			result.operator.id = processStep.processor.__id__;
+			result.operator.name = processStep.processor.fullName;
+			otherValue = processStep.rejectionReason || '';
+			if (processingStep.isRejected) {
+				result.hasOnlySystemicReasons = false;
+				result.rejectionReasons.push(
+					rejectionReason('processingStep', ['other'], otherValue, '')
+				);
+			}
+			return true;
+		});
+		//rejectionReasons -> data
+		if (businessProcess.dataForms.rejectReason) {
+			result.hasOnlySystemicReasons = false;
+			result.rejectionReasons.push(rejectionReason('data', ['other'],
+				businessProcess.dataForms.rejectReason, ''));
+		}
+		//rejectionReasons -> requirementUploads
+		businessProcess.requirementUploads.applicable.forEach(function (requirementUpload) {
+			if (requirementUpload.status !== 'invalid') return;
+			types = [];
+			otherValue = '';
+			requirementUpload.rejectReasonTypes.forEach(function (type) {
+				if (type === 'other') {
+					result.hasOnlySystemicReasons = false;
+					otherValue = requirementUpload.rejectReasonMemo;
+				}
+				types.push(type);
+			});
+			result.rejectionReasons.push(
+				rejectionReason('requirementUpload', types, otherValue
+				  , requirementUpload.__id__.slice(requirementUpload.__id__.indexOf('/')))
+			);
+		});
+		//rejectionReasons -> paymentReceiptUploads
+		businessProcess.paymentReceiptUploads.applicable.forEach(function (paymentReceiptUpload) {
+			if (paymentReceiptUpload.status !== 'invalid') return;
+			result.hasOnlySystemicReasons = false;
+			result.rejectionReasons.push(
+				rejectionReason('paymentReceipt', ['other'], paymentReceiptUpload.rejectReasonMemo
+				  , paymentReceiptUpload.__id__.slice(paymentReceiptUpload.__id__.indexOf('/')))
+			);
+		});
+
+		result.date = Number(new db.DateTime());
```

Comment:
`Number(new db.Date())`

we don't want whole `DateTime` - we want date without time on the view, and we want to group by it later on.